File: Task_Angular_v19/src/app/modules/employee/services/employee.service.ts
import { HttpClient, HttpHeaders } from '@angular/common/http';
import { Injectable } from '@angular/core';
import { Observable, throwError } from 'rxjs';
import { StorageService } from '../../../auth/services/storage/storage.service';

@Injectable({
  providedIn: 'root'
})
export class EmployeeService {

  constructor(private http: HttpClient) { }

  getTasksByUserId(): Observable<any> {
    const userId = StorageService.getUserId();
    if (!userId) {
      console.error('User ID is not available');
      return throwError(() => new Error('User ID is not set.'));
    }
    return this.http.get(`http://localhost:8080/api/employee/tasks/${userId}`, httpOptions);
  }

  getTaskById(id: number): Observable<any> {
    return this.http.get('http://localhost:8080/api/employee/task/' + id, httpOptions);
  }

  getCommentsById(id: number): Observable<any> {
    return this.http.get('http://localhost:8080/api/employee/task/' + id + '/comments', httpOptions);
  }

  createComment(taskId: number, content: string): Observable<any> {
    const params_ = {
      taskId: taskId,
      postedBy:StorageService.getUserId() ?? ''
    };
    return this.http.post('http://localhost:8080/api/employee/task/comment', content, {
      params: params_,
      headers: new HttpHeaders({
        'Content-Type': 'application/json',
        'Authorization': `Bearer ${StorageService.getToken()}`
      }),
      withCredentials: true
    });
  }

  updateTask(id: number, status:string): Observable<any> {
    return this.http.get('http://localhost:8080/api/employee/task/' + id + '/' + status, httpOptions);
  }
}

const httpOptions = {
  headers: new HttpHeaders({
    'Content-Type': 'application/json',
    'Authorization': `Bearer ${StorageService.getToken()}`
  }),
  withCredentials: true
};

File: TaskManagement/src/main/java/com/finalproject/TaskManagement/services/employee/EmployeeService.java
package com.finalproject.TaskManagement.services.employee;

import com.finalproject.TaskManagement.dtos.CommentDTO;
import com.finalproject.TaskManagement.dtos.TaskDTO;

import java.util.List;

/**
 * Сервис для работы с задачами и комментариями для сотрудников.
 */
public interface EmployeeService {

    /**
     * Получает задачи, связанные с пользователем по его идентификатору.
     *
     * @param id идентификатор пользователя
     * @return список задач в виде DTO
     */
    List<TaskDTO> getTasksByUserId(Long id);

    /**
     * Обновляет статус задачи.
     *
     * @param id идентификатор задачи
     * @param status новый статус задачи
     * @return обновленная задача в виде DTO
     */
    TaskDTO updateTask(Long id, String status);

    /**
     * Получает задачу по идентификатору.
     *
     * @param id идентификатор задачи
     * @return задача в виде DTO
     */
    TaskDTO getTaskById(Long id);

    /**
     * Создает комментарий для задачи.
     *
     * @param taskId  идентификатор задачи
     * @param postedBy идентификатор пользователя, опубликовавшего комментарий
     * @param content содержимое комментария
     * @return созданный комментарий в виде DTO
     */
    CommentDTO createComment(Long taskId, Long postedBy, String content);

    /**
     * Получает комментарии для задачи.
     *
     * @param taskId идентификатор задачи
     * @return список комментариев в виде DTO
     */
    List<CommentDTO> getCommentsByTask(Long taskId);
}
